examples/vision/video_transformers.py

```diff
@@ -0,0 +1,408 @@
+exammple. In this example, we will be using a Transformer-based
+([Vaswani et al.](https://arxiv.org/abs/1706.03762)) model to classify videos. You can follow
+[this book chapter](https://livebook.manning.com/book/deep-learning-with-python-second-edition/chapter-11)
+in case you need an introduction to Transformers with code. After reading this
+example, you will know how to develop a hybrid Transformer-based models for video
+classification that operate on CNN feature maps.
+
+This example requires TensorFlow 2.5 or higher, as well as TensorFlow Docs, which can be
+installed using the following command:
+"""
+
+"""shell
+pip install -q git+https://github.com/tensorflow/docs
+"""
+
+"""
+## Data collection
+
+As done in the [predecessor](https://keras.io/examples/vision/video_classification/) to
+this example, we will be using a subsampled version of the [UCF101 dataset](https://www.crcv.ucf.edu/data/UCF101.php)
+which is a known benchmark dataset. In case you want to operate on a larger subsample or
+even the entire dataset, please refer to 
+[this notebook](https://colab.research.google.com/github/sayakpaul/Action-Recognition-in-TensorFlow/blob/main/Data_Preparation_UCF101.ipynb).
+"""
+
+"""shell
+wget -q https://git.io/JGc31 -O ucf101_top5.tar.gz
+tar xf ucf101_top5.tar.gz
+"""
+
+"""
+## Setup
+"""
+
+from tensorflow_docs.vis import embed
+from tensorflow.keras import layers
+from tensorflow import keras
+
+import tensorflow as tf
+import pandas as pd
+import numpy as np
+import imageio
+import cv2
+import os
+
+"""
+## Define hyperparameters
+"""
+
+MAX_SEQ_LENGTH = 20
+NUM_FEATURES = 1024
+IMG_SIZE = 128
+
+EPOCHS = 5
+
+"""
+## Data preparation
+
+We will mostly be following the same data preparation steps in this example except for
+the following changes:
+
+* Reducing the image sizes to 128 instead of 224 to speed up computation. 
+* Instead of using a pre-trained [InceptionV3](https://arxiv.org/abs/1512.00567) network,
+we will be using a pre-trained
+[DenseNet121](http://openaccess.thecvf.com/content_cvpr_2017/papers/Huang_Densely_Connected_Convolutional_CVPR_2017_paper.pdf)
+for feature extraction. 
+* We will directly be padding the shorter videos to the `MAX_SEQ_LENGTH`. 
+
+First, let's load up the
+[DataFrames](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html). 
+
+**Note**: In case you did not follow the data preparation steps in the previous example
+mentioned at the beginning, you are strongly advised to do that first before proceeding. 
+"""
+
+train_df = pd.read_csv("train.csv")
+test_df = pd.read_csv("test.csv")
+
+print(f"Total videos for training: {len(train_df)}")
+print(f"Total videos for testing: {len(test_df)}")
+
+# The following two methods are taken from this tutorial:
+# https://www.tensorflow.org/hub/tutorials/action_recognition_with_tf_hub
+
+
+def crop_center_square(frame):
```

Comment:
You could use https://keras.io/api/layers/preprocessing_layers/image_preprocessing/center_crop/